src/Workspaces/Core/Portable/Workspace/Solution/DocumentState_LinkedFileReuse.cs

```diff
@@ -69,8 +69,11 @@ public DocumentState UpdateTextAndTreeContents(
         // We only need to look one deep here as we'll pull that tree source forward to our level.  If another link is
         // later added to us, it will do the same thing.
         var originalTreeSource = this.TreeSource;
-        if (originalTreeSource is LinkedFileReuseTreeAndVersionSource linkedFileTreeAndVersionSource)
+        if (originalTreeSource is LinkedFileReuseTreeAndVersionSource linkedFileTreeAndVersionSource
+            && !linkedFileTreeAndVersionSource.TryGetValue(out var _))
```

Comment:
this def needs docs.

Reply:
alternatively, if you are able to get the value from the linkedSource, why not pass it along instead?  And if you can't, you defer to the original tree source it points to.

Also, i think we should doc .OriginalTreeSource.  Mention that it's there to produce a long chain if we haven't computed the intermediary steps.  But if we have, we'll use that value instead.

Reply:
Moved code around a bit and added some comments.